packages/grid/_modules_/grid/hooks/features/virtualization/useGridVirtualScroller.tsx

```diff
@@ -288,6 +293,12 @@ export const useGridVirtualScroller = (props: UseGridVirtualScrollerProps) => {
           selected={selectionLookup[id] !== undefined}
           index={currentPage.range.firstRowIndex + nextRenderContext.firstRowIndex! + i}
           containerWidth={availableSpace}
+          className={
+            isEmptySpace &&
+            isLastRow &&
+            !rootProps.autoHeight &&
+            gridClasses['row--lastBeforeEmpty']
```

Comment:
Could we keep the calculation of the class name inside the `useUtilityClasses` of the `GridRow`? You can pass the input values you need as props. This allows the users to opt-out from this feature by providing a custom row through the slots.